src/main/java/org/openstreetmap/atlas/geography/atlas/change/ChangeAtlas.java

```diff
@@ -52,6 +64,30 @@
     private transient Long numberOfPoints;
     private transient Long numberOfRelations;
 
+    // Computing relations in ChangeAtlas is very expensive, so we cache them here.
+    private transient Map<Long, ChangeRelation> relationsCache;
```

Comment:
This is very clever. Leveraging the `atlas` immutability is a great idea. We really shouldn't have been recomputing relations on the fly every time one was requested from the `ChangeAtlas`.

Reply:
Yep, but this is also a risk with the memory pressure. This will duplicate memory needed to make this work.